LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,270 @@
+
+        public void PrintDeploymentMessage(string buildName, List<BuildRegionParams> regionConfigurations)
+        {
+            Console.WriteLine($"Starting deployment {buildName} for titleId, regions  {string.Join(", ", regionConfigurations.Select(x => x.Region))}");
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithProcessBasedServerResponse>> CreateBuildWithProcessBasedServer(CreateBuildWithProcessBasedServerRequest request)
+        {
+            PrintDeploymentMessage(request.BuildName, request.RegionConfigurations);
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithProcessBasedServerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithManagedContainerResponse>> CreateBuildWithManagedContainer(CreateBuildWithManagedContainerRequest request)
+        {
+            PrintDeploymentMessage(request.BuildName, request.RegionConfigurations);
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithManagedContainerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithCustomContainerResponse>> CreateBuildWithCustomContainer(CreateBuildWithCustomContainerRequest request)
+        {
+            PrintDeploymentMessage(request.BuildName, request.RegionConfigurations);
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithCustomContainerAsync(request);
+        }
+
+        public List<PlayFab.MultiplayerModels.Port> PortMapping()
+        {
+            var ports = new List<Port>();
+
+            foreach (var portList in settings.PortMappingsList)
+            {
+                ports.AddRange(portList?.Select(x => new Port()
+                {
+                    Name = x.GamePort.Name,
+                    Num = settings.RunContainer ? x.GamePort.Number : 0,
+                    Protocol = (ProtocolType)Enum.Parse(typeof(ProtocolType), x.GamePort.Protocol)
+                }).ToList());
+            }
+
+            return ports;
+        }
+
+        public string GetAssetFileNameFromPath(string filePath)
+        {
+            return Path.GetFileName(filePath);
+        }
+
+        public async Task<PlayFabResult<GetAssetDownloadUrlResponse>> FileExistsInBlob(string filename)
+        {
+            GetAssetDownloadUrlRequest downloadRequest = new GetAssetDownloadUrlRequest() { FileName = filename };
+
+            return await PlayFabMultiplayerAPI.GetAssetDownloadUrlAsync(downloadRequest);
+        }
+
+        public void PrintError(dynamic error)
+        {
+            if (error != null && error.ErrorMessage != null)
+            {
+                Console.WriteLine($"{error.ErrorMessage}");
+            }
+        }
+
+        public async Task CheckAssetFiles(string filename)
+        {
+           
+            var filevalidator = FileExistsInBlob(filename);
+
+            if (filevalidator.Result.Result == null)
+            {
+                GetAssetUploadUrlRequest request1 = new() { FileName = filename };
```

Comment:
make sure to replace new() to new GetAssetUploadUrlRequest () to fix an issue with test failure